### PR TITLE
Auth/issue 2 signup login implementation

### PR DESCRIPTION
**Pull request by Tam9927**

### CODE REVIEW

File: Backend/Controller/auth.controller.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+async function registerUser(req, res) {
+  try {
+    const data = await authService.register(req.body);
+    if (data) {
+      let accesstoken = jwt.sign(
+        { username: data.username },
+        process.env.ACCESS_TOKEN_SECRET,
+        {
+          algorithm: "HS256",
```
Comment (edge555) [inline Backend/Controller/auth.controller.js]: keep the algorithm name in env file

File: Backend/.env [hunk 2]
```diff
@@ -0,0 +1,6 @@
+APP_NAME='your app name'
+APP_PORT='port number'
+SALTROUND='salt for hashing'
+
+ACCESS_TOKEN_SECRET=swsh23hjddnns
+ACCESS_TOKEN_LIFE=12000000
```
Comment (edge555) [inline Backend/.env]: don't expose env file publically, keep this in gitignore

File: Backend/Services/auth.service.js [hunk 3]
```diff
@@ -14,6 +14,11 @@ async function register(user) {
   }
 
   try {
+    //const id = crypto.randomUUID();
+    const salt = await bcrypt.genSalt(10);
```
Comment (edge555) [inline Backend/Services/auth.service.js:28]: put salting value in env

File: Backend/utils/user.utils.js [hunk 4]
```diff
@@ -35,6 +35,20 @@ function generateToken(username) {
...
+    .cookie('jwt', null, {
+        expiresIn: new Date(Date.now())
+    }, 
+    {httpOnly: true})
+    .json({
+        success: true,
+        message: "Logged Out"
+    });
+
+}
```
Comment (edge555) [inline Backend/utils/user.utils.js:50]: Deleting the cookie should be enough. refer to this link for better understanding, https://stackoverflow.com/questions/37959945/how-to-destroy-jwt-tokens-on-logout

**Review by rafaelragib — CHANGES_REQUESTED** [review]

File: Backend/Controller/auth.controller.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+require("dotenv").config();
+const authService = require("../Services/auth.service");
+const userUtils = require("../utils/user.utils");
+const jwt = require("jsonwebtoken");
+
+async function registerUser(req, res) {
+  try {
+    const data = await authService.register(req.body);
```
[the remark this answers is not in the document]
Comment (Tam9927) [reply Backend/Controller/auth.controller.js:12]: It is handled as an error in the catch block of registerUser

File: Backend/utils/user.utils.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+function checkUsernameValid(username) {
+  const usernameValidCheck = /[^A-Za-z0-9]/;
+  if (usernameValidCheck.test(username)) {
+    return false;
+  }
+  return true;
+}
```
Comment (rafaelragib) [inline Backend/utils/user.utils.js:59]: the function can be written in one line using ternary operator in the return statement

File: Backend/utils/user.utils.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+function checkPasswordValid(password) {
+  if (password.length < 6) {
+    return false;
+  }
+  return true;
+}
```
Comment (rafaelragib) [inline Backend/utils/user.utils.js:66]: the function can be written in one line using ternary operator in the return statement

File: Backend/utils/user.utils.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+function checkEmailValid(email) {
+  if (validator.validate(email)) {
+    return true;
+  } else {
+    return false;
+  }
+}
```
Comment (rafaelragib) [inline Backend/utils/user.utils.js:74]: the function can be written in one line using ternary operator in the return statement

File: Backend/Controller/auth.controller.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+const jwt = require("jsonwebtoken");
+
+
+async function registerUser(req, res) {
+  try { 
+
+    const data = await authService.register(req.body);
+    const status = data?.status;
+    const message = data?.message;
+
```
Comment (rafaelragib) [inline Backend/Controller/auth.controller.js:15]: why is this optional?

File: Backend/Controller/auth.controller.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+    console.log(err);
+    res.status(500).send("An error occurred");
+  }
+}
+
+async function loginUser(req, res) {
+  try {
+    const data = await authService.loginUser(req.body);
+    const status = data?.status;
+    const message=data?.message
```
Comment (rafaelragib) [inline Backend/Controller/auth.controller.js:40]: why is this optional?

File: Backend/Services/auth.service.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
+"use strict"
+
+const userUtils = require("../utils/user.utils");
+const UserRepository = require("../repository/user.repository");
```
Comment (rafaelragib) [inline Backend/Services/auth.service.js:4]: unused module

**Review by rafaelragib — CHANGES_REQUESTED** [review]
Fix the issues and merge the PR

File: Backend/Controller/auth.controller.js (hunk 10 above)
Comment (Tam9927) [reply Backend/Controller/auth.controller.js:40]: This came because of an accidental merge issue with eslint

**Pull request merged**